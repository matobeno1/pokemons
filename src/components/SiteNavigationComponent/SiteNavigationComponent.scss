.site-navigation-component {
  padding: 1rem;
  position: fixed;
  font-weight: 600;
  width: 100%;
  --scale: 1;
  --pokeball-size: calc(var(--scale) * 1.5rem);
  --text-size: calc(var(--scale) * 1.1rem);

  @include sm() {
    --scale: 1.1;
  }

  @include lg() {
    --scale: 1.3;
  }

  &__back {
    display: flex;
    align-items: center;
  }

  &__image {
    position: absolute;
    height: var(--pokeball-size);
    width: var(--pokeball-size);
  }

  &__back-text {
    margin-left: calc(var(--pokeball-size) * 1.1);
    color: var(--color-text);
    font-size: var(--text-size);
  }
}
