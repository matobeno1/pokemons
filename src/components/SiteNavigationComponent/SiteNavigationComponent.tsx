import { FunctionComponent } from "react";
import Link from "next/link";
import Image from "next/image";
import { createClassNames } from "../../bem";

const classNames = createClassNames("site-navigation-component");

export const SiteNavigationComponent: FunctionComponent = () => (
    <nav className={classNames()}>
        <Link href="/" passHref>
            <div role="link" className={classNames("back")}>
                <span className={classNames("image")}>
                    <Image src="/pokeball.svg" layout="fill" alt="logo" />
                </span>
                <span className={classNames("back-text")}>
                    Pok√©dex
                </span>
            </div>
        </Link>
    </nav>
);
