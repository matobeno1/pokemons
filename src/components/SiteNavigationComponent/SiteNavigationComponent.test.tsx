import { act, create } from "react-test-renderer";
import { SiteNavigationComponent } from "./SiteNavigationComponent";

describe("SiteNavigationComponent", () => {
    it("snapshots", async () => {
        let root: ReturnType<typeof create> | undefined;
        act(() => {
            root = create(<SiteNavigationComponent />);
        });
        expect(root?.toJSON()).toMatchSnapshot();
    });
});
