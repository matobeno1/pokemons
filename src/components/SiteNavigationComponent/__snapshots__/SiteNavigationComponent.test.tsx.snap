// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteNavigationComponent snapshots 1`] = `
<nav
  className="site-navigation-component"
>
  <link
    href="/"
    passHref={true}
  >
    <a
      className="site-navigation-component__back"
      role="link"
    >
      <span
        className="site-navigation-component__image"
      >
        <img
          alt="logo"
          layout="fill"
          src="/pokeball.svg"
        />
      </span>
      <span
        className="site-navigation-component__back-text"
      >
        Pok√©dex
      </span>
    </a>
  </link>
</nav>
`;
